Library/Homebrew/download_strategy.rb

```diff
@@ -559,6 +559,27 @@ def curl(*args, **options)
   end
 end
 
+# Strategy for downloading a file using brewed curl.
+#
+# @api public
+class BrewedCurlDownloadStrategy < CurlDownloadStrategy
+  private
+
+  def _fetch(url:, resolved_url:, timeout:)
+    ohai "Downloading from #{resolved_url}" if url != resolved_url
+
+    raise BrewedCurlDownloadStrategyError, url unless Formula["curl"].any_version_installed?
+
+    if Homebrew::EnvConfig.no_insecure_redirect? &&
+       url.start_with?("https://") && !resolved_url.start_with?("https://")
+      $stderr.puts "HTTPS to HTTP redirect detected & HOMEBREW_NO_INSECURE_REDIRECT is set."
+      raise CurlDownloadStrategyError, url
+    end
+
+    curl_download resolved_url, to: temporary_path, timeout: timeout, use_brewed_curl: true
```

Comment:
Is this the only bit that differs from the other `_fetch`? If so, it may be worth having a `_curl_download` or `_curl_download_args` or similar method that can be override to DRY up a bit.

Reply:
Yes it's the only difference. I refactored it as asked;